File: FlickrFindr/app/src/main/java/com/example/flickrfindr/ui/fragments/PhotoFragment.java
package com.example.flickrfindr.ui.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.flickrfindr.api.ImageViewExtension;
import com.example.flickrfindr.databinding.PhotoFragmentBinding;

public class PhotoFragment extends Fragment {
    static String TAG = PhotoFragment.class.getName();
    private PhotoFragmentBinding binding;

    public PhotoFragment() {
        this.binding = null;
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState) {
        binding = PhotoFragmentBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (getArguments() != null) {
            ImageViewExtension.loadImage(binding.flickrPhoto, getArguments().getString("photo_url"));
            binding.outsideImgView.setOnClickListener(l -> requireActivity().getSupportFragmentManager().popBackStack());
        }
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }
}

File: FlickrFindr/app/src/main/java/com/example/flickrfindr/ui/fragments/PhotoFragment.kt
package com.example.flickrfindr.ui.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.flickrfindr.databinding.PhotoFragmentBinding
import loadImage

class PhotoFragment : Fragment() {
    private var binding: PhotoFragmentBinding? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = PhotoFragmentBinding.inflate(inflater, container, false)
        return binding?.root
    }

    override fun onViewCreated(view: View, bundle: Bundle?) {
        super.onViewCreated(view, bundle)
        if (arguments != null) {
            binding?.flickrPhoto?.loadImage(arguments?.getString("photo_url").toString())
            binding?.outsideImgView?.setOnClickListener {
                activity?.supportFragmentManager?.popBackStack()
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        binding = null
    }

    companion object {
        val TAG: String = PhotoFragment::class.java.name
    }
}
